scipy/stats/unuran/unuran_wrapper.pyx.templ

```diff
@@ -0,0 +1,869 @@
+            out_cont = np.empty(N, dtype=np.float64)
+            self._rvs_cont(out_cont)
+            if size is None:
+                return out_cont[0]
+            return np.asarray(out_cont).reshape(size)
+        elif unur_distr_is_discr(unur_get_distr(self.rng)):
+            out_discr = np.empty(N, dtype=np.int32)
+            self._rvs_discr(out_discr)
+            if size is None:
+                return out_discr[0]
+            return np.asarray(out_discr).reshape(size)
+        else:
+            raise NotImplementedError("only univariate continuous and "
+                                      "discrete distributions supported")
+
+    def __dealloc__(self):
+        if self.rng != NULL:
+            unur_free(self.rng)
+        if self.urng != NULL:
+            unur_urng_free(self.urng)
+
+
+cdef class TransformedDensityRejection(Method):
+    r"""
+    Transformed Density Rejection (TDR) Method.
+
+    TDR is an acceptance/rejection method that uses the concavity of a
+    transformed density to construct hat function and squeezes automatically.
+    Most universal algorithms are very slow compared to algorithms that are
+    specialized to that distribution. Algorithms that are fast have a slow
+    setup and require large tables. The aim of this universal method is to
+    provide an algorithm that is not too slow and needs only a short setup.
+    This method can be applied to continuous unimodal distributions with
+    T-concave density function. See [1]_ and [2]_ for more details.
+
+    Parameters
+    ----------
+    pdf : callable
+        PDF of the distribution. The signature of the PDF is expected to be:
+        ``def pdf(x: float, *params) -> float``. i.e. the PDF should accept
+        a Python float and the parameters and return a Python float. It need
+        not integrate to 1 i.e. the PDF doesn't need to be normalized.
+    dpdf : callable
+        Derivative of the PDF w.r.t the variate. Must have the same signature
+        as the PDF.
+    mode : float, optional
+        Mode of the distribution. It is optional and only required if
+        `use_mode` is ``True``.
+    center : float, optional
+        Center (``loc``) of the distribution. It is optional and only
+        required if `use_center` is ``True``.
+    params : tuple, optional
+        Parameters that the PDF and DPDF take. Default is an empty tuple.
+    domain : list or tuple of length 2, optional
+        The support of the distribution. Must not contain `nan` values.
+        Default is ``None``. When ``None``, the support is assumed to be
+        :math:`(-\infty, \infty)`.
+    c : float, optional
+        The transformation function. See Notes for the allowed values for
+        this parameter and their interpretation.
+    cpoints : int, optional
+        The number of construction points. Default is 30.
+    variant : str, optional
+        The variant to use. Available options are:
+
+          * 'ps': PS variant (Default)
```

Comment:
this is from the UNU.RAN documentation:
GW: squeezes between construction points
PS: squeezes proportional to hat function (Default)
IA: same as variant PS but uses a compositon method with “immediate acceptance” in the region below the squeeze.

Reply:
i saw that you put that description below. in that case, I would just write Available variantes are 'ps', 'ia' and 'gw', see Notes for details

Reply:
> i saw that you put that description below. in that case, I would just write Available variantes are 'ps', 'ia' and 'gw', see Notes for details

Oops, I should have done that! Sorry. I will push the change in the next commit.

Reply:
Done.

Reply:
I think you forgot to mark this as resolved. I have changed this in bcd404ca7d2b7f0a199a37320fd6f998a10e6cf5